targets/sommelier

```diff
@@ -0,0 +1,74 @@
+#!/bin/sh -e 
+# Copyright (c) 2021 The crouton Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+REQUIRES='core'
+PROVIDES='x11'
+DESCRIPTION='Wayland backend running in a Chromium OS window.'
+HOSTBIN='startsommelier'
+CHROOTBIN='runsommelier'
+CHROOTETC='sommelierrc'
+. "${TARGETSDIR:="$PWD"}/common"
+
+### Append to prepare.sh:
+XMETHOD="${XMETHOD:-sommelier}"
+
+# Download the versioned sommelier
+sommelier_version="fef7f012df0e5ec865f94255c10503ec24d9a370"
+urlbase="https://chromium.googlesource.com/chromiumos/platform2/+archive/$sommelier_version/vm_tools"
+
+# Download the corresponding virtwl.h
+kernel_version="7f7629cc680bf3c22d547fb5333e521af58c6c9e"
+urlvirtwl="https://chromium.googlesource.com/chromiumos/third_party/kernel/+/$kernel_version/include/uapi/linux/virtwl.h?format=TEXT" 
+
+DUMMYBUILDTMP="`mktemp -d crouton-sommelier.XXXXXX --tmpdir=/tmp`"
+
+addtrap "rm -rf --one-file-system '$DUMMYBUILDTMP'"
+
+echo "Download ..." 1>&2
+
+superinsecure=''
+if release -le wheezy; then
+     # Wheezy doesn't have the right CA in ca-certificates that validates x.org
+     superinsecure='--no-check-certificate'
+fi
+wget $superinsecure -O "$SOMMELIERBUILDTMP/sommelier.tar.gz" \
+     "$urlbase/sommelier.tar.gz"
+
+mkdir "$SOMMELIERBUILDTMP/linux"
+wget $superinsecure -O- "$urlvirtwl" \
+    | base64 --decode > "$SOMMELIERBUILDTMP/linux/virtwl.h"
+
+install xwayland weston xorg
+
+install --minimal --asdeps pkg-config libwayland-dev libgbm-dev gcc \
+    libx11-xcb-dev libsystemd-dev libxcb-composite0-dev \
+    libxkbcommon-dev libxrender-dev libxtst-dev libpixman-1-dev \
+    ninja-build meson cmake libdrm-dev build-essential
+
+# Compilation subshell
+(
+    cd "$SOMMELIERBUILDTMP"
+    # -m prevents "time stamp is in the future" spam
+    tar -xzmf sommelier.tar.gz
```

Comment:
When we start `Xwayland` without `-rootless` it is a fullscreen window in the screen that currently has the focus. In principle, this should be possible since when I connect an external monitor which is differently shaped, the window was not fullscreen any more.  However, I did not manage resize it or specify a size. I have inadequate wayland-foo and the window is borderless :-(

I can try to create the argument processing patch.

Reply:
Regarding `sommelierrc`, I suppose it should then be OK to pull it from git-versioned google sources instead of creating a duplicate.

In that case, does it go in `/usr/local/etc/sommelierrc` or `/etc/crouton/sommelierc`?

Reply:
commit 41dfd2e7fed7f2e006c7c95d29a5cd0f09e18b90 
installs ChromiumOS's `sommelierrc`